Decred Wallet/Features/Wallets/WalletSettingsViewController.swift

```diff
@@ -60,6 +60,36 @@ class WalletSettingsViewController: UIViewController {
         self.incomingTxAlertButton.setTitle(newSetting.localizedString, for: .normal)
     }
     
+     @IBAction func rescanBlockchain(_ sender: Any) {
+        if SyncManager.shared.isSyncing {
+            Utils.showBanner(in: self.view, type: .error, text: LocalizedStrings.errorSyncInProgress)
+        } else if !WalletLoader.shared.multiWallet.isConnectedToDecredNetwork() {
+            Utils.showBanner(in: self.view, type: .error, text: LocalizedStrings.notConnected)
+        } else if SyncManager.shared.isRescanning {
+            Utils.showBanner(in: self.view, type: .error, text: LocalizedStrings.errorRescanInProgress)
+        } else {
+            // rescan blockchain
+            self.showOkAlert(message: LocalizedStrings.rescanConfirm,
+                                 title: LocalizedStrings.rescanBlockchain,
+                                 onPressOk: self.rescanBlocks,
+                                 addCancelAction: true)
+        }
+    }
+        
+    func rescanBlocks() {
+        do {
+            try WalletLoader.shared.multiWallet.rescanBlocks(self.wallet.id_)
+            DispatchQueue.main.async {
+                Utils.showBanner(in: self.view, type: .success, text: LocalizedStrings.rescanProgressNotification)
+            }
```

Comment:
I don't think it is necessary to explicitly run the banner display code in the main thread, the `rescanBlocks` function don't seem to be called from a background thread.